pkg/render/intrusion_detection.go

```diff
@@ -1297,9 +1320,48 @@ func (c *intrusionDetectionComponent) adAPIService() *corev1.Service {
 	}
 }
 
-func (c *intrusionDetectionComponent) adAPIDeployment() *appsv1.Deployment {
-	adAPIStorageVolumePath := "/storage"
-	adAPIStorageVolumeName := "volume-storage"
+func (c *intrusionDetectionComponent) adPersistentVolumeClaim(configureADStorage bool) *corev1.PersistentVolumeClaim {
```

Comment:
Does it really matter what the value of the boolean is here? Outside of this func you delete the spec if the value is false, and otherwise you create it.

Reply:
it matters only during a creation (ie. if it is an object that needs to be added in the `Objects()` method), since it needs to select whether to use `emptyDir()` or  set the volume to refer to the StorageClass

for deletion it would always get the value of false

Reply:
ended up removing this, ^ invalid comment now, was thinking of the deployment